pkg/ipcache/types.go

```diff
@@ -95,7 +94,10 @@ func (m *resourceInfo) merge(info IPMetadata, src source.Source) bool {
 		changed = m.endpointFlags != info
 		m.endpointFlags = info
 	default:
-		log.Errorf("BUG: Invalid IPMetadata passed to ipinfo.merge(): %+v", info)
+		logger.Error(
+			"BUG: Invalid IPMetadata passed to ipinfo.merge()",
+			logfields.Info, info,
```

Comment:
Maybe `fmt.Sprintf("%+v", info)` unless that's the default of slog representations